Upgrade to Gradle 8.14.3

wjglerum opened the pull request:
Bump Gradle to the latest version, see https://docs.gradle.org/8.14.3/release-notes.html

And also fixes some deprecations while we are at it:
- Removes deprecated usage of `Task.getProject()`
- Removes deprecated usage of `Project.exec()`
- Removes deprecated Kotlin compiler options

Started this cause I was trying out the gradle 9 release candidates and some things are deprecated/removed.

sergej-koscejev commented:
The commit message says '8.14.2' but upgrades to 8.14.3.

According to Gradle docs, ExecOperations has been available since Gradle 6.0 so you should be free to use it, the plugin will only run on something like 7.x or newer anyway (or maybe even 8.x), as far as I know.

wjglerum commented:
> The commit message says '8.14.2' but upgrades to 8.14.3.

whoops will fix that 😄 

> According to Gradle docs, ExecOperations has been available since Gradle 6.0 so you should be free to use it, the plugin will only run on something like 7.x or newer anyway (or maybe even 8.x), as far as I know.

Alright will try to get rid of the deprecations too then 👍

wjglerum commented:
> According to Gradle docs, ExecOperations has been available since Gradle 6.0 so you should be free to use it, the plugin will only run on something like 7.x or newer anyway (or maybe even 8.x), as far as I know.

@sergej-koscejev done, leveraged property injection to inject the `ExecOperations` where needed, see https://docs.gradle.org/current/userguide/service_injection.html#property_injection

sergej-koscejev commented on src/main/kotlin/de/itemis/mps/gradle/RunAntScript.kt:
```diff
@@ -70,30 +71,30 @@ open class RunAntScript : DefaultTask() {
         val targets = if (incremental) { targets - "clean" } else { targets }
 
-        project.runAnt(executable, project.rootDir, allArgs + "-buildfile" + project.file(script).toString() + targets,
+        project.runAnt(execOperations, executable, project.rootDir, allArgs + "-buildfile" + project.file(script).toString() + targets,
```
We shouldn't need Project here anymore, it was only used to call javaexec.

wjglerum replied:
That's also what I thought, but we actually do use it to get some properties from the project in the internal method

sergej-koscejev commented on src/main/kotlin/de/itemis/mps/gradle/downloadJBR/Plugin.kt:
```diff
@@ -21,7 +22,11 @@ open class DownloadJbrConfiguration @Inject constructor(objects: ObjectFactory)
+abstract class DownloadJbrProjectPlugin : Plugin<Project> {
+
+    @get:Inject
+    abstract val execOperations: ExecOperations
```
```suggestion
    protected abstract val execOperations: ExecOperations
```

sergej-koscejev commented on src/main/kotlin/de/itemis/mps/gradle/RunAntScript.kt:
```diff
@@ -28,6 +27,8 @@ open class RunAntScript : DefaultTask() {
     @Optional @Input
     var executable: Any? = null
+    @get:Inject
+    abstract val execOperations: ExecOperations
```
```suggestion
    protected abstract val execOperations: ExecOperations
```

sergej-koscejev commented on src/main/kotlin/de/itemis/mps/gradle/runmigrations/Plugin.kt:
```diff
@@ -205,7 +209,7 @@ open class RunMigrationsMpsProjectPlugin : Plugin<Project> {
 
                         runAnt(
-                            extension.javaExec, temporaryDir, args = listOf(),
+                            execOperations,extension.javaExec, temporaryDir, args = listOf(),
```
```suggestion
                            execOperations, extension.javaExec, temporaryDir, args = listOf(),
```

sergej-koscejev commented:
Looks good, please add a (short) changelog entry and clean up the commit history.

wjglerum committed:
Update to Gradle 8.14.3

wjglerum committed:
Migrate deprecated kotlin compiler options

wjglerum committed:
Migrate deprecated `Project#exec` to use injected `ExecOperations` instead

Leverage property based injection to avoid breaking backward compatibility with the constructor.  See https://docs.gradle.org/current/userguide/service_injection.html#property_injection

wjglerum committed:
Remove another usage of deprecated `Task#getProject`

Introduces a loglevel property for MPS tasks with a convention to use the gradle loglevel. Removes the redundant utility methods and just adds the arguments inline.

wjglerum committed:
Release version 1.29.3

wjglerum commented:
> Looks good, please add a (short) changelog entry and clean up the commit history.

Done! ✅
